rest/replication_api_test.go

```diff
@@ -597,6 +601,12 @@ func TestReplicationRebalancePull(t *testing.T) {
 	assert.Equal(t, "remoteRT", docABC2Body2["source"])
 	docDEF2Body2 := activeRT2.getDoc(docDEF2)
 	assert.Equal(t, "remoteRT", docDEF2Body2["source"])
+
+	// Identify rebalanced replication, and validate stats are preserved
+	replicationStatuses := activeRT2.GetReplicationStatuses("?localOnly=true")
+	assert.Equal(t, 1, len(replicationStatuses))
+	rebalancedReplicationID := replicationStatuses[0].ID
+	activeRT2.assertReplicationStat(rebalancedReplicationID, "docs_read", 2)
```

Comment:
This test is failing in Jenkins.

Reply:
The failing test is a race on replication rebalance.  On replication stop we force publication of replication status, which can be used to initialize on replication start.  Stop and Start aren't guaranteed to occur in that order, though - it depends when the cfg update reaches the old/new nodes via DCP.  

This looks like another justification for replication status being published whenever we write checkpoints, to ensure that the initial status matches the replication starting point.  I'm considering whether we should just store the relevant stats inside the checkpoint document itself.  To support rollback I expect that would require storing in both the local and remote checkpoint documents.  This would result in an extra 200-300 bytes when we write checkpoints, which I don't think is excessive given the checkpoint persistence frequency.

Reply:
I'm on board with this, and wouldn't be too difficult to add into the checkpoint doc either.

The downside I can think of would be N+1 ops to read stats for N replications (one to get the cluster config to generate checkpoint IDs for each, and N ops to get the checkpoint doc).